core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -779,7 +784,8 @@ object IO extends Serializable {
     override def tag = Tags.Supervise
   }
 
-  final class Terminate private[IO] (val causes: List[Throwable]) extends IO[Nothing, Nothing] {
+  final class Terminate private[IO] (val defect: Option[Throwable], val causes: List[Throwable])
```

Comment:
We should turn `defect` into `Throwable`, rather than `Option[Throwable]`. See below comment.